src/overlays/fbdemos/ovl_fbdemo_wipe5/z_fbdemo_wipe5.c

```diff
@@ -16,37 +18,133 @@ void TransitionWipe5_SetColor(void* thisx, u32 color);
+            break;
+        default:
+            this->unk_8 = 0.2f;
+            break;
+    }
+    switch (this->unk_11) {
+        case 0:
+            this->primColor.rgba = 0x000000FF;
+            break;
+        case 1:
+            this->primColor.rgba = 0xA0A0A0FF;
+            break;
+        default:
+            this->primColor.rgba = 0xA0A0A0FF;
+            break;
+    }
+}
+
+void* TransitionWipe5_Init(void* thisx) {
+    TransitionWipe5* this = THIS;
+
+    bzero(this, sizeof(TransitionWipe5));
+    return this;
+}
+
+void TransitionWipe5_Destroy(void* thisx) {
+}
+
+void TransitionWipe5_Update(void* thisx, s32 updateRate) {
+    TransitionWipe5* this = THIS;
+
+    if (!this->unk_10) {
+        this->isDone = true;
+    } else {
+        this->unk_C = this->unk_C + ((this->unk_8 * 3.0f) / updateRate);
+        if (this->unk_C >= 1.0f) {
+            this->unk_C = 1.0f;
+            this->isDone = true;
+        }
+    }
+}
+
+void TransitionWipe5_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    TransitionWipe5* this = THIS;
+    s32 alpha;
+    s32 sp50 = D_801FBB90;
+    s32 width;
+    s32 height;
 
-extern u16 D_0F000000[];
+    alpha = ((1.0f - this->unk_C) * 255.0f);
+    width = gScreenWidth;
+    height = gScreenHeight;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_fbdemo_wipe5/TransitionWipe5_Start.s")
+    gDPPipeSync(gfx++);
+    gSPLoadUcodeL(gfx++, gspS2DEX2_fifo);
+    if (!this->unk_10) {
+        gDPSetOtherMode(gfx++,
+                        G_AD_PATTERN | G_CD_MAGICSQ | G_CK_NONE | G_TC_CONV | G_TF_POINT | G_TT_NONE | G_TL_TILE |
+                            G_TD_CLAMP | G_TP_NONE | G_CYC_COPY | G_PM_NPRIMITIVE,
+                        G_AC_NONE | G_ZS_PIXEL | G_RM_NOOP | G_RM_NOOP2);
+        func_8014116C(&gfx, D_0F000000, sp50, width, height, 1);
+    } else {
+        if (alpha == 0xFF) {
+            gDPSetOtherMode(gfx++,
+                            G_AD_DISABLE | G_CD_DISABLE | G_CK_NONE | G_TC_FILT | G_TF_POINT | G_TT_NONE | G_TL_TILE |
+                                G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
+                            G_AC_NONE | G_ZS_PRIM | G_RM_OPA_SURF | G_RM_OPA_SURF2);
+        } else {
+            gDPSetOtherMode(gfx++,
+                            G_AD_NOISE | G_CD_NOISE | G_CK_NONE | G_TC_FILT | G_TF_POINT | G_TT_NONE | G_TL_TILE |
+                                G_TD_CLAMP | G_TP_NONE | G_CYC_1CYCLE | G_PM_NPRIMITIVE,
+                            G_AC_NONE | G_ZS_PRIM | G_RM_CLD_SURF | G_RM_CLD_SURF2);
+        }
+        gDPSetEnvColor(gfx++, 255, 255, 255, alpha);
+        gDPSetCombineLERP(gfx++, TEXEL0, 0, ENVIRONMENT, 0, 0, 0, 0, ENVIRONMENT, TEXEL0, 0, ENVIRONMENT, 0, 0, 0, 0,
+                          ENVIRONMENT);
+        func_8014116C(&gfx, sp50, D_0F000000, width, height, 0);
+    }
+    gDPPipeSync(gfx++);
+    gSPLoadUcode(gfx++, SysUcode_GetUCode(), SysUcode_GetUCodeData());
+    *gfxP = gfx;
```

Comment:
I'd be good to space out the pipesyncs and microcode loads a bit more.. There's a few other tweaks here, but I'll do it myself later.